src/Microsoft.DotNet.Interactive.CSharpProject/RoslynWorkspaceUtilities.cs

```diff
@@ -0,0 +1,389 @@
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Host;
+using Microsoft.CodeAnalysis.Text;
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using static Microsoft.DotNet.Interactive.CSharpProject.BuildCacheFileUtilities;
+
+namespace Microsoft.DotNet.Interactive.CSharpProject;
+
+public static class RoslynWorkspaceUtilities
+{
+    public class ProjectItem
+    {
+        public string ItemSpec { get; }
+        public IReadOnlyDictionary<string, string> Metadata { get; }
+
+        internal ProjectItem()
+        {
+//            ItemSpec = taskItem.ItemSpec;
+//            Metadata = taskItem.MetadataNames.Cast<string>().ToDictionary(x => x, x => taskItem.GetMetadata(x));
+        }
+    }
+
+    public class BuildDataResults
+    {
+        public BuildProjectData Data { get; set; }
+
+        public Microsoft.CodeAnalysis.Workspace Workspace { get; set; }
+
+        public bool Succeeded { get; set; }
+
+        public string cacheFile { get; set; }
+
+        public CSharpParseOptions CSharpParseOptions { get; set; }
+
+        // TODO: Set this value
+        public string ProjectFilePath { get; set; }
+
+        public Dictionary<string, ProjectItem[]> Items;
+    }
+
+    internal static BuildDataResults GetResultsFromCacheFile(string cacheFilePath)
+    {
+        Microsoft.CodeAnalysis.Workspace ws = null;
+        BuildProjectData buildProjectData = null;
+        CSharpParseOptions cSharpParseOptions = null;
+
+        try
+        {
+            if (!File.Exists(cacheFilePath))
+            {
+                return null;
+            }
+
+            string fileContent = File.ReadAllText(cacheFilePath);
+
+            PopulateBuildProjectData(fileContent, out buildProjectData);
+
+            ws = GetWorkspace(buildProjectData, true);
+
+            cSharpParseOptions = CreateCSharpParseOptions(buildProjectData);
+
+            return new BuildDataResults()
+            {
+                Data = buildProjectData,
+                ProjectFilePath = cacheFilePath,
+                cacheFile = cacheFilePath + cacheFilenameSuffix,
+                Workspace = ws,
+                Succeeded = true,
+                CSharpParseOptions = cSharpParseOptions
+            };
+        } catch (Exception)
+        {
+            return null;
+        }
+    }
+
+    internal static BuildDataResults GetResultsFromProjectFile(string csprojFilePath)
+    {
+        if (!File.Exists(csprojFilePath))
+        {
+            return null;
+        }
+
+        var cacheFilePath = csprojFilePath + BuildCacheFileUtilities.cacheFilenameSuffix;
+
+        return GetResultsFromCacheFile(cacheFilePath);
+    }
+
+    internal static void PopulateBuildProjectData(string fileContent, out BuildProjectData buildProjectData)
+    {
+        buildProjectData = new BuildProjectData
```

Comment:
Typically it's preferable for these property initializations to be the defaults for the class itself.

Reply:
done in the last commit